src/v2/view-builder/views/phone/EnrollAuthenticatorPhoneView.js

```diff
@@ -1,9 +1,148 @@
 import BaseForm from '../../internals/BaseForm';
 import BaseFactorView from '../shared/BaseFactorView';
+import CountryUtil from '../../../../util/CountryUtil';
 
 const Body = BaseForm.extend({
+
+  className: 'phone-authenticator-enroll',
+
+  title () {
+    return loc('oie.phone.enroll.title', 'login');
+  },
+
+  subtitle () {
+    return loc('oie.phone.enroll.subtitle', 'login');
+  },
+
+  _changeView ({ changed }) {
+    // Update the button label and value..
+    const btn = this.el.querySelector('.o-form-button-bar .button-primary');
+    const phoneField = this.el.querySelector('.phone-authenticator-enroll__phone');
+    const extensionField = this.el.querySelector('.phone-authenticator-enroll__phone-ext');
+
+    const voiceSelected = changed['authenticator.methodType'] === 'voice';
+    const smsSelected = changed['authenticator.methodType'] === 'sms';
+    const smsBtnText = loc('oie.phone.enroll.smsButton', 'login');
+    const voiceBtnText = loc('oie.phone.enroll.voiceButton', 'login');
+
+    if (voiceSelected) {
+      btn.innerText = voiceBtnText;
+      btn.value = voiceBtnText;
+      if (!phoneField.classList.contains('phone-authenticator-enroll__phone--small')) {
+        phoneField.classList.add('phone-authenticator-enroll__phone--small');
+      }
+      extensionField.classList.remove('hidden');
+    }
+
+    if (smsSelected) {
+      btn.innerText = smsBtnText;
+      btn.value = smsBtnText;
+      phoneField.classList.remove('phone-authenticator-enroll__phone--small');
+      if (!extensionField.classList.contains('hidden')) {
+        extensionField.classList.add('hidden');
+      }
+    }
+  },
+
+  _changeCountry ({ changed }) {
+    const countryCodeField = this.el.querySelector('.phone-authenticator-enroll__phone-code');
+    countryCodeField.innerText = `+${CountryUtil.getCallingCodeForCountry(changed.country)}`;
+  },
+
+  save () {
+    return loc('oie.phone.enroll.smsButton', 'login');
+  },
+
+  initialize () {
+    const argArr = Array.prototype.slice.call(arguments);
+    const { currentViewState } = argArr[0];
+
+    const countryUISchema = {
+      'label-top': true,
+      label: 'Country',
+      type: 'select',
+      options: CountryUtil.getCountries(),
+      name: 'country',
+    };
+
+    // Create an input group- serves as a display
+    const phoneNumberUISchema = {
+      label: 'Phone number',
+      type: 'group',
+      modelType: 'string',
+      'label-top': true,
+      className : 'phone-authenticator-enroll__phone',
+      input: [
+        {
+          type: 'label',
+          label: '+1',
+          className: 'phone-authenticator-enroll__phone-code',
+        },
+        Object.assign({}, currentViewState.uiSchema[2]),
+      ]
+    };
+
+    const extensionUISchema = {
+      label: 'Extension',
+      type: 'text',
+      className: 'phone-authenticator-enroll__phone-ext hidden',
+      'label-top': true,
+      name: 'extension',
+    };
+
+    // Add country select input
```

Comment:
- override `getUISchema` to insert more ui schemas. search for examples in `src/v2/`
- search for desired input by name instead of index like `currentViewState.uiSchema[2]`